### PR TITLE
Aktivities in platforms website

### PR DESCRIPTION
**Pull request by MustafaTRKYLMZ**

### CODE REVIEW

File: apps/api/src/api/activity/content-types/activity/schema.json [hunk 1]
```diff
@@ -107,6 +113,27 @@
...
+      },
+      "type": "boolean"
+    },
+    "forLotus": {
+      "pluginOptions": {
+        "i18n": {
+          "localized": true
+        }
+      },
+      "type": "boolean"
```
Comment (7alip) [inline apps/api/src/api/activity/content-types/activity/schema.json:136]: Why we have both `platforms` relation and `forLotus`, `forKunsthalte` booleans.

`forPlatform` boolean approach was an alternative way to relation. We should chose one of them. I prefer using `platforms` relation approach.

File: packages/ui/src/data/schemas/activity.ts [hunk 2]
```diff
@@ -22,6 +24,8 @@ export const activityFields: FormFields<Activity> = [
 
   { name: 'content', isRequired: true, type: 'markdown' },
   { name: 'image', type: 'file', isRequired: true },
+  { name: 'forKunsthalte', type: 'boolean',label: 'Publish to Kunsthalte', },
+  { name: 'forLotus', type: 'boolean',label:'Publish to Lotus vd Media' },
```
Comment (7alip) [inline packages/ui/src/data/schemas/activity.ts:28]: If we use `platforms` relation, these fields should be replaced with `platforms` multi select.

File: packages/types/src/activity.ts [hunk 3]
```diff
@@ -6,6 +6,8 @@ import { Tag } from './tag'
 
 type ActivityBase = StrapiEntityBase & {
   date: string
+  forKunsthalte:boolean
+  forLotus:boolean
 }
 
 type ActivityRelation = {
```
Comment (7alip) [inline packages/types/src/activity.ts:13]: Did you forget to add `platforms` relation as type?

Comment (MustafaTRKYLMZ) [reply packages/types/src/activity.ts:13]: > Did you forget to add `platforms` relation as type?
I don't understand that. I thought that forKunsthalte and forLotus are enought.

Comment (7alip) [reply packages/types/src/activity.ts:13]: In the [old issue](https://github.com/wsvvrijheid/monorepo/issues/403) I meantioned `forPlatform` approach but in my opinion `platforms` approach is better as we discussed in the [new issue](https://github.com/wsvvrijheid/monorepo/issues/758).

`platforms` relation is more flexible and in the future, if there are more platforms we can easily update platform relation.

File: apps/foundation/pages/platforms/index.tsx [hunk 4]
```diff
@@ -12,6 +12,8 @@ import { HomePlatform, Layout } from '../../components'
 type PlatformsProps = InferGetStaticPropsType<typeof getStaticProps>
 
 const Platforms: FC<PlatformsProps> = ({ title, platforms }) => {
+  console.log('platforms BBBBBB', platforms)
```
Comment (7alip) [inline apps/foundation/pages/platforms/index.tsx:15]: Console could be removed

File: apps/kunsthalte/pages/activities/index.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    locale,
+    page,
+    filters: { platforms: { id: { $in: [1] } } },
+  })
+
+  const { data, isLoading } = activitiesQuery
+
+  const pagination = data?.meta?.pagination
+  const activities = data?.data || []
+  console.log('aktivities :::', activities)
```
Comment (7alip) [inline apps/kunsthalte/pages/activities/index.tsx:50]: Console could be removed

File: packages/types/src/activity.ts [hunk 6]
```diff
@@ -13,12 +14,14 @@ type ActivityRelation = {
...
   image?: UploadFile
   localizations?: Array<Activity>
+  platforms?: Array<Platform>
 }
 
 type ActivityRelationInput = {
   categories?: number
   tags?: Array<number>
   image: File
+  platforms?: number
```
Comment (7alip) [inline packages/types/src/activity.ts:24]:
```suggestion
  platforms?: number[]
```

File: packages/ui/src/admin/ModelForm/utils.ts [hunk 7]
```diff
@@ -152,6 +153,13 @@ export const useDefaultValues = <T extends StrapiModel>(
               value: c.id.toString(),
             })) || []
           break
+        case 'platforms':
+          defaults.platforms =
+            activityModel?.platforms?.map(c => ({
```
Comment (7alip) [inline packages/ui/src/admin/ModelForm/utils.ts:158]:
```suggestion
            activityModel?.platforms?.map(p => ({
```

**Review by 7alip — CHANGES_REQUESTED** [review]
Thanks, great!

**Comment by 7alip** [comment]
@MustafaTRKYLMZ You can merge after finishing the small changes

File: apps/foundation/pages/platforms/index.tsx (hunk 4 above)
Comment (7alip) [inline apps/foundation/pages/platforms/index.tsx:15]: Is the console still there?

File: apps/kunsthalte/pages/activities/index.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+const args: RequestCollectionArgs = {
+  url: 'api/activities',
+  sort: ['date:desc'],
+  filters: {
+    approvalStatus: { $eq: 'approved' },
+    platforms: { id: { $in: [1] } },
+  },
+  populate: ['image', 'platforms'],
+  fields: ['title', 'description', 'slug'],
```
Comment (7alip) [inline apps/kunsthalte/pages/activities/index.tsx:30]: It's better not to use fields here. We might need date or other data in the future for cards

File: apps/foundation/pages/platforms/index.tsx (hunk 4 above)
Comment (MustafaTRKYLMZ) [reply apps/foundation/pages/platforms/index.tsx:15]: I deleted. But Maybe I forgot.

File: apps/kunsthalte/pages/activities/index.tsx (hunk 8 above)
Comment (MustafaTRKYLMZ) [reply apps/kunsthalte/pages/activities/index.tsx:30]: It was before. If it is not nessesary, I can delete.

File: apps/foundation/pages/platforms/index.tsx (hunk 4 above)
Comment (7alip) [reply apps/foundation/pages/platforms/index.tsx:15]: It seems you forgot to delete it.

File: apps/kunsthalte/pages/activities/index.tsx (hunk 8 above)
Comment (7alip) [reply apps/kunsthalte/pages/activities/index.tsx:30]: Not necessary for now. It might be necessary when we update the Card

**Pull request merged**